src/Core/src/Handlers/HybridWebView/HybridWebView.ts

```diff
@@ -0,0 +1,216 @@
+ * communicate between the web view and the .NET host application.
+ * 
+ * The JavaScript file is generated from TypeScript and should not be modified
+ * directly. To make changes, modify the TypeScript file and then recompile it.
+ */
+
+/*
+ * Extend the global objects to include the interfaces that are used by the
+ * HybridWebView and some operating systems and/or browser hosts.
+ */
+interface External {
+    receiveMessage: (message: any) => void;
+}
+interface Window {
+    chrome?: {
+        webview?: {
+            addEventListener: (event: string, handler: (arg: any) => void) => void;
+            postMessage: (message: any) => void;
+        };
+    };
+    webkit?: {
+        messageHandlers?: {
+            webwindowinterop?: {
+                postMessage: (message: any) => void;
+            };
+        };
+    };
+
+    // Declare the global object that we have added on Android.
+    hybridWebViewHost?: {
+        sendMessage: (message: string) => void;
+    };
+}
+
+/*
+ * The following interfaces define the shape of the messages that are sent between
+ * the web view and the .NET host application.
+ */
+interface JSInvokeMethodData {
+    MethodName: string;
+    ParamValues?: string[];
+}
+interface JSInvokeError {
+    Name?: string;
+    Message: string;
+    StackTrace?: string;
+}
+interface DotNetInvokeResult {
+    IsJson: boolean;
+    Result: any;
+}
+
+(() => {
+
+    /*
+     * Initialize the HybridWebView messaging system.
+     * This method is called once when the page is loaded.
+     */
+    function initHybridWebView() {
+        function dispatchHybridWebViewMessage(message: any) {
+            const event = new CustomEvent('HybridWebViewMessageReceived', { detail: { message: message } });
+            window.dispatchEvent(event);
+        }
+
+        if (window.chrome?.webview?.addEventListener) {
+            // Windows WebView2
+            window.chrome.webview.addEventListener('message', (arg: any) => {
+                dispatchHybridWebViewMessage(arg.data);
+            });
+        } else if (window.webkit?.messageHandlers?.webwindowinterop) {
+            // iOS and MacCatalyst WKWebView
+            // @ts-ignore - We are extending the global object here
+            window.external = {
+                receiveMessage: (message: any) => {
+                    dispatchHybridWebViewMessage(message);
+                },
+            };
+        } else {
+            // Android WebView
+            window.addEventListener('message', (arg: any) => {
+                dispatchHybridWebViewMessage(arg.data);
+            });
+        }
+    }
+
+    /*
+     * Send a message to the .NET host application.
+     * The message is sent as a string with the following format: `<type>|<message>`.
+     */
+    function sendMessageToDotNet(type: string, message: string) {
+        const messageToSend = type + '|' + message;
+
+        if (window.chrome && window.chrome.webview) {
```

Comment:
Nit: You can use optional chaining here, like is done in `initHybridWebView()`:

```suggestion
        if (window?.chrome?.webview) {
```